pkg/ccl/backupccl/testdata/backup-restore/restore-protect-spans

```diff
@@ -0,0 +1,55 @@
+
+# tickle the unprotected timestamp bug
+new-server name=s1 disable-tenant
+----
+
+exec-sql
+CREATE DATABASE d;
+USE d;
+CREATE TABLE foo (i INT PRIMARY KEY, s STRING);
+INSERT INTO foo VALUES (1, 'x'),(2,'y');
+----
+
+exec-sql
+BACKUP INTO 'nodelocal://1/full_cluster_backup/';
+----
+
+
+exec-sql
+SET CLUSTER SETTING bulkio.ingest.flush_delay='20s';
+----
+
+exec-sql
+SET CLUSTER SETTING spanconfig.reconciliation_job.check_interval='1s';
+----
+
+exec-sql
+ALTER DATABASE defaultdb CONFIGURE ZONE USING gc.ttlseconds = 1;
+----
+
+
+exec-sql
+SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_flow';
+----
+
+restore expect-pausepoint tag=a
+RESTORE TABLE foo FROM LATEST IN 'nodelocal://1/full_cluster_backup/' WITH into_db='defaultdb';
+----
+job paused at pausepoint
+
+# ensure foo has adopted the ttl
+sleep ms=5000
```

Comment:
When I manually conduct the test in demo, the ttl was reconciled to the restored table, so i’m unsure why this job succeeds.

Reply:
Random thoughts:

- You could move the sleep to before the restore and then add a step here which prints out the span configurations, just to make sure the gc.ttl has been reconciled. 

- Does the error require that GC has actually run on the relevant ranges? Perhaps that isn't happening in this test?

Reply:
It could also be the pts cache timing: `SET CLUSTER SETTING kv.protectedts.poll_interval = '1s'`

Reply:
Yeah just to second steven's comments I think this test will be better suited as a non-DD test that uses some of the utility methods in `utils_test.go`. Such as `waitForReplicaFieldToBeSet` where you can wait for the replica to have the spanconfig you expect it to have. `TestExcludeDataFromBackupDoesNotHoldupGC` is an example where this util is used.

Reply:
ah, yeah. that was my initial strategy, but a some of those unfamiliar utils were not quite working on the offline table. I'll take a closer look today and try to modify them to read directly from the system tables.

Reply:
After thinking more about this, I’ve realized the original bug is quite challenging to repro in a unit test. If we do indeed trust the PTS system, is it sufficient to merely check that  during restore, we properly add a PTS and clean it up after the restore is done?

  To directly repro the bug, we need to induce and complete a gc job on a range processing a really slow addstable request. But does this repro really add more coverage? I think we can trust that the PTS system will prevent the gc threshold from advancing on the range.

Reply:
see example of the alternative approach in `TestProtectRestoreSpans`